main.tf

```diff
@@ -0,0 +1,172 @@
+# ------------------------------------------------------------------------------
+# AWS
+# ------------------------------------------------------------------------------
+data "aws_region" "current" {}
+
+# ------------------------------------------------------------------------------
+# Cloudwatch
+# ------------------------------------------------------------------------------
+resource "aws_cloudwatch_log_group" "main" {
+  name              = "${var.name_prefix}"
+  retention_in_days = "${var.log_retention_in_days}"
+  tags              = "${var.tags}"
+}
+
+# ------------------------------------------------------------------------------
+# IAM - Task execution role, needed to pull ECR images etc.
+# ------------------------------------------------------------------------------
+resource "aws_iam_role" "execution" {
+  name               = "${var.name_prefix}-task-execution-role"
+  assume_role_policy = "${data.aws_iam_policy_document.task_assume.json}"
+}
+
+resource "aws_iam_role_policy" "task_execution" {
+  name   = "${var.name_prefix}-task-execution"
+  role   = "${aws_iam_role.execution.id}"
+  policy = "${data.aws_iam_policy_document.task_ecr_permissions.json}"
+}
+
+# ------------------------------------------------------------------------------
+# IAM - Task role, basic. Users of the module will append policies to this role
+# when they use the module. S3, Dynamo permissions etc etc.
+# ------------------------------------------------------------------------------
+resource "aws_iam_role" "task" {
+  name               = "${var.name_prefix}-task-role"
+  assume_role_policy = "${data.aws_iam_policy_document.task_assume.json}"
+}
+
+resource "aws_iam_role_policy" "log_agent" {
+  name   = "${var.name_prefix}-log-permissions"
+  role   = "${aws_iam_role.task.id}"
+  policy = "${data.aws_iam_policy_document.task_permissions.json}"
+}
+
+# ------------------------------------------------------------------------------
+# Security groups
+# ------------------------------------------------------------------------------
+resource "aws_security_group" "ecs_service" {
+  vpc_id      = "${var.vpc_id}"
+  name        = "${var.name_prefix}-ecs-service-sg"
+  description = "Fargate service security group"
+  tags        = "${merge(var.tags, map("Name", "${var.name_prefix}-sg"))}"
+}
+
+resource "aws_security_group_rule" "ingress_service" {
```

Comment:
I don't remember why we had `icmp` traffic allowed but I don't think its needed. When running with public IP in the default VPC any outsider can ping our service which we don't want.